src/algebra/algebra/restrict_scalars.lean

```diff
@@ -116,16 +150,12 @@ instance [I : ring A] : ring (restrict_scalars R S A) := I
 instance [I : comm_semiring A] : comm_semiring (restrict_scalars R S A) := I
 instance [I : comm_ring A] : comm_ring (restrict_scalars R S A) := I
 
-variables [comm_semiring S] [semiring A]
-
-instance restrict_scalars.algebra_orig [I : algebra S A] : algebra S (restrict_scalars R S A) := I
-
-variables [algebra S A]
+variables [semiring A]
 
-/-- Tautological `S`-algebra isomorphism `restrict_scalars R S A ≃ₐ[S] A`. -/
-def restrict_scalars.alg_equiv : restrict_scalars R S A ≃ₐ[S] A := alg_equiv.refl
+/-- Tautological ring isomorphism `restrict_scalars R S A ≃+* A`. -/
+def restrict_scalars.ring_equiv : restrict_scalars R S A ≃+* A := ring_equiv.refl _
```

Comment:
Can you add the lemma that says what this does to `algebra_map R (restrict_scalars R S)`?

Reply:
Ah, good catch, I've added that one!!